sockeye/transformer.py

```diff
@@ -381,15 +381,12 @@ def cast(self, dtype):
         super().cast(dtype)
 
     def hybrid_forward(self, F, x):
-        # (length)
-        x = F.squeeze(F.slice(x, begin=(0, None, 0), end=(1, None, 1)))
-        # (length, 1)
-        # TODO: use F.contrib.arange_like with MXNET 1.6.0
-        length_array = F.cast(F.contrib.index_array(x, axes=(1,)), dtype=self._dtype)
+        # Shape: (length, 1)
+        length_array = F.contrib.arange_like(x, axis=1)
         # matrix with lower triangle and main diagonal set to 0, upper triangle set to 1
         # Shape: (length, length)
-        bias = F.broadcast_greater(F.reshape(length_array, shape=(1, -1)),
-                                   length_array)
+        bias = F.broadcast_greater(F.expand_dims(length_array, axis=0),
+                                   F.expand_dims(length_array, axis=1))
         bias = bias * -C.LARGE_VALUES[self._dtype]
```

Comment:
[Curiosity only] Why is this necessary? I read the docs and saw that for np.float16 it will move it to some multiple of 32 and that for np.float32 it will move to 1.0e8. Is the goal to keep the np.float32 masked logits intact and round them up when using np.float16?

Reply:
If training in FP16 you want to avoid overflowing the numerical range, therefore the max value for FP16 is different than for FP32.